src/monattack.cpp

```diff
@@ -2255,6 +2255,165 @@ void mattack::tentacle(monster *z, int index)
     g->u.check_dead_state();
 }
 
+
+void mattack::pull_enemy(monster *z, int index)
+{
+    int t;
+    int t2 = 0;
+    Creature *target = z->attack_target();
+    monster *zz = dynamic_cast<monster*>(target);
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 3
+            || rl_dist( z->pos(), target->pos() ) <= 1 || !z->sees(*target, t)) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    std::vector<tripoint> line = line_to( z->pos(), target->pos(), t, t2 );
+    bool seen = g->u.sees( *z );
+
+    z->reset_special(index); // Reset timer
+    z->moves -= 150;
+
+    for (auto &i : line){
+        const ter_t &terrain = g->m.ter_at(i);
+        //Player can't be pulled though bars
+        if (terrain.movecost == 0 ){
+            add_msg( _("The %s flings its arms at you, but they bounce off the %s"), z->name().c_str(), terrain.name.c_str() );
+            return;
+        }
+    }
+    bool uncanny = target->uncanny_dodge();
+
+    if( uncanny || dodge_check(z, target) ) {
+        z->moves -=200;
+        auto msg_type = foe == &g->u ? m_warning : m_info;
+        if( foe != nullptr ) {
+            foe->add_msg_player_or_npc( msg_type, _("The %s's arms fly out at you, but you dodge!"),
+                                                    _("The %s's arms fly out at <npcname>, but they dodge!"),
+                                        z->name().c_str() );
+
+        }
+        else if( !uncanny ) {
+            target->on_dodge( z, z->type->melee_skill * 2 );
+        }
+        else {
+            foe->add_msg_player_or_npc( msg_type, _("The %s's arms fly out at you, but them miss you!"),
+                                        _("The %s's arms fly out at <npcname>, but they miss!"),
+                                z->name().c_str() );
+        }
+        return;
+    }
+
+    const int dir = g->m.coord_to_angle( z->posx(), z->posy(), target->posx(), target->posy() ); //Orientation of where you'll end up
+    tileray tdir(dir);
+    int range = (z->type->melee_sides * z->type->melee_dice) / 10;
+    tripoint pt = target->pos();
+    while (range > 0) {
+        if( foe != nullptr ) {
+        tdir.advance();
+        pt.x = z->posx() + tdir.dx();
+        pt.y = z->posy() + tdir.dy();
```

Comment:
The part below doesn't properly check if the location is free. Most of the time it works fine, consider this situation:
`Zz@` where `Z` is the grappler, `z` is any zed and `@` is the player. This will put the player on the zed.
With monsters, the effect is also a debug message.

Test it out by spawning a debug horde of friendly grapplers and hostile grapplers and have them fight to the death.